StorageTests/TestDeferredSqlite.swift

```diff
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+import XCTest
+
+
+class TestDeferredSqlite: XCTestCase {
+    func testCancelling() {
+        let files = MockFiles()
+        let db = SwiftData(filename: files.getAndEnsureDirectory(relativeDir: "test.db", error: nil)!)
+        let expectation = self.expectationWithDescription("Wait")
+
+        let deferred = DeferredSqliteOperation(block: { (connection, err) -> Int? in
+            XCTFail("This should be cancelled before hitting this")
```

Comment:
Isn't this actually dictated by timing? That is, this operation, once started, will run on some background queue. A few lines below we cancel it, but it might have already completed, no?